index.js

```diff
@@ -10,8 +10,69 @@ const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
  * @param compHex filter for complementary color hex code
  * @returns Promise
  */
-const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+
+const fetchColors = async ({ name, hex, compName, compHex }) => {
+  // use fetch api to retrieve json entries
+  try {
+    // set variable "response" equal to the response object returned from fetch
+    const response = await fetch(COLORS);
+    // if something foes wrong, put up this error message
+    if (!response.ok) {
+      throw new Error("aiya");
+    }
+    // set variable "data" equal to the response body in json format
+    const data = await response.json();
+    
+    // create empty variable to set answer equal to
+    let result;
+
+    // now use switch statements to fork the function if certain inputs are detected
+    switch (true) {
+      // follow this path if "name" parameter exists
+      case (name != null):
+        // make the name parameter case insensitive
+        const name_input = name.toLowerCase();
+        // filter results by variable "name_input"  
+        result = data.filter((color) => color.name.toLowerCase() == name_input)
+        break;
+      // follow this path if "hex" parameter exists
+      case (hex != null):
+        // filter results by hex parameter  
+        result = data.filter((color) => color.hex == hex)
+        break;
+      // follow this path if "compName" parameter exists
+      case (compName != null):
+        // make the compName parameter case insensitive
+        const compName_input = compName.toLowerCase();
+        // filter to see if each color's comp array includes the compName parameter in its name element
+        result = data.filter((color) => {
+          // find the first instance in the comp list that contains compName_input string 
+          const find_compName = color.comp.find((color) => color.name.toLowerCase().includes(compName_input))
```

Comment:
Remember to DRY your code. The string comparison could be moved to its own function so if the requirement changed to be case sensitive you only need to do it in one place.